lua/entities/gmod_wire_pod.lua

```diff
@@ -456,163 +490,184 @@ local function fixupangle(angle)
 	return angle
 end
 
+local function recacheOutput(entity, selfTbl, oname, value)
+	if selfTbl["Last" .. oname] == value then return end
+	selfTbl["Last" .. oname] = value
+	WireLib.TriggerOutput(entity, oname, value)
+end
+
 function ENT:Think()
-	if (self:HasPly() and self.Activated) then
-		local ply = self:GetPly()
+	local selfTbl = self:GetTable()
+	local ply = self:GetPly()
+
+	if ply and selfTbl.Activated then
 		local pod = self:GetPod()
 
 		-- Tracing
-		local trace = util.TraceLine( { start = ply:GetShootPos(), endpos = ply:GetShootPos() + ply:GetAimVector() * 9999999999, filter = { ply, pod } } )
+		local shootPos = ply:GetShootPos()
+		local aimVector = ply:GetAimVector()
+		local trace = util.TraceLine({ start = shootPos, endpos = shootPos + aimVector * 9999999999, filter = { ply, pod } })
 		local distance
-		if (self:HasPod()) then distance = trace.HitPos:Distance( pod:GetPos() ) else distance = trace.HitPos:Distance( ply:GetShootPos() ) end
+		local hitPos = trace.HitPos
+		if pod then distance = hitPos:Distance(pod:GetPos()) else distance = hitPos:Distance(shootPos) end
 
-		if (trace.Hit) then
+		if trace.Hit then
 			-- Position
-			WireLib.TriggerOutput( self, "X", trace.HitPos.x )
-			WireLib.TriggerOutput( self, "Y", trace.HitPos.y )
-			WireLib.TriggerOutput( self, "Z", trace.HitPos.z )
-			WireLib.TriggerOutput( self, "AimPos", trace.HitPos )
-			WireLib.TriggerOutput( self, "Distance", distance )
-			self.VPos = trace.HitPos
+			recacheOutput(self, selfTbl, "X", hitPos.x)
+			recacheOutput(self, selfTbl, "Y", hitPos.y)
+			recacheOutput(self, selfTbl, "Z", hitPos.z)
+			recacheOutput(self, selfTbl, "AimPos", hitPos)
+			recacheOutput(self, selfTbl, "Distance", distance)
```

Comment:
Doesn't TriggerOuput already do nothing if the output port matches the trigger value? I think this added logic is redundant.

Reply:
It is redundant, but it's significantly faster than going through the rest of the TriggerOutput logic to reach that point if you profile it.

Reply:
Can we just improve TriggerOutput instead?

Reply:
I think it does have a lot of hacky stuff in it iirc

Reply:
Can we compare with the stored output instead of storing it in the entity table?
e.g. self.Outputs.X.value. Slightly less efficient but this isn't a heavy performance section.

Reply:
Wiremod core just needs a rewrite in its entirety :P

Reply:
Outputs should be a class, each type having their own trigger function, yeah.